src/pages/ScholarshipForm.jsx

```diff
@@ -42,26 +133,54 @@ function ScholarshipForm() {
     <Container maxWidth="md">
       <form>
         <h1>Submit a Scholarship</h1>
-        {requiredTextField('name', 'Scholarship Name')}
+        <div>
+          {requiredTextField(
+            'name',
+            'Scholarship Name',
+            formFieldErrors.name,
+            'Please enter a name'
+          )}
+        </div>
         <DatePicker
           id="deadline"
           label="Deadline"
+          error={formFieldErrors.deadline}
+          helperText="Please enter a valid date"
           value={formFieldStates.deadline}
           onChange={(date) =>
             setFormFieldStates({ ...formFieldStates, deadline: date })
           }
         />
-        {requiredTextField('description', 'Description')}
-        {requiredTextField('website', 'Website')}
+        <div>
+          {requiredTextField(
+            'description',
+            'Description',
+            formFieldErrors.description,
+            'Please enter a description',
+            true
+          )}
+        </div>
+        <div>
+          {requiredTextField(
+            'website',
+            'Website',
+            formFieldErrors.website,
+            'Please enter a valid website'
+          )}
+        </div>
         <ScholarshipAmountField
+          typeError={formFieldErrors.amountType}
+          minAmountError={formFieldErrors.minAmount}
+          maxAmountError={formFieldErrors.maxAmount}
+          helperText={amountHelperText}
           amountType={formFieldStates.amountType}
           minAmount={formFieldStates.minAmount}
           maxAmount={formFieldStates.maxAmount}
           onTypeChange={updateFn('amountType')}
           updateAmount={updateAmount}
         />
         <div>
-          <Button variant="contained" color="primary">
+          <Button variant="contained" color="primary" onClick={validateFields}>
```

Comment:
Have you tried setting the form's `onSubmit` handler and making this an `<input type="submit" value="Submit" component={Button} color="primary"/>` element?  I'm looking at the [React forms docs](https://reactjs.org/docs/forms.html)

Reply:
I'm curious if that would help highlight the required fields when the user tries to submit instead of having to check every single required field.

Reply:
Since the `onSubmit` handler would receive all form values, I also wonder if there's a way to avoid passing in `onChange` handlers and instead just pass in validation functions and have the component itself worry about when to surface an error.

Reply:
ok so I tried this and it does highlight the required fields, and also checks depending on the type. Should I do it this way or keep and manage the states. One thing I'm not sure about yet is how to make it work for the amount fields, specifically the range

Reply:
Form validation is supposed to help avoid doing common validations you'll find across forms. I suggest using the **Submit** button and utilizing [built-in form validation](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#Using_built-in_form_validation). This can cover these use cases of ours:
- Min/Max amount values
- Required fields set
- TextField `type`'s like `url`. Or alternatively `pattern`

For styling, the elements will match the `:invalid` CSS pseudoclass.

Btw, if you want to prevent the redirect/refresh on form submission you'll need to add `event.preventDefault()` in the `onSubmit` handler

Reply:
I do think you might need to still keep track of state though.

Reply:
It's up to you ultimately but I would opt for whatever is easiest.

Reply:
I'm going to punt this for #209 to take care of. There's other potential solutions that might be better